onnxruntime/core/providers/cuda/nn/conv.cc

```diff
@@ -212,45 +254,42 @@ Status Conv<T>::UpdateState(OpKernelContext* context, bool bias_expected) const
       int cudnn_conv_algo = cuda_ep->GetCudnnConvAlgo();
       ORT_ENFORCE(cudnn_conv_algo > -1 && cudnn_conv_algo < 3, "cudnn_conv_algo should be 0, 1 or 2, but got ", cudnn_conv_algo);
       switch (cudnn_conv_algo) {
-        case 0:
+        case 0: {
+          static constexpr int num_algos = CUDNN_CONVOLUTION_FWD_ALGO_COUNT;
+          size_t max_ws_size = getMaxWorkspaceSize(s_, kAllAlgos, num_algos);
+          IAllocatorUniquePtr<void> algo_search_workspace = GetScratchBuffer<void>(max_ws_size);
```

Comment:
+1 for this change - users have HEURISTIC and DEFAULT need not pay this allocation cost !